app/src/main/java/org/openobservatory/ooniprobe/receiver/TestRunBroadRequestReceiver.java

```diff
@@ -0,0 +1,122 @@
+        String value = intent.getStringExtra("value");
+        switch (key) {
+            case TestAsyncTask.START:
+                listener.onStart(service);
+                break;
+            case TestAsyncTask.RUN:
+                listener.onRun(value);
+                break;
+            case TestAsyncTask.PRG:
+                List<AbstractSuite> previousTestSuits =
+                        service.task.testSuites.subList(0, service.task.testSuites.indexOf(service.task.currentSuite));
+                int previousTestProgress = (int) Stats.of(Lists.transform(
+                        previousTestSuits,
+                        input -> input.getTestList(preferenceManager).length * 100
+                )).sum();
+                int previousTestRuntime = (int) Stats.of(Lists.transform(
+                        previousTestSuits,
+                        input -> input.getRuntime(preferenceManager)
+                )).sum();
+                int prgs = Integer.parseInt(value);
+                int currentTestRuntime = service.task.currentSuite.getRuntime(preferenceManager);
+                int currentTestMax = service.task.currentSuite.getTestList(preferenceManager).length * 100;
+                double timeLeft = runtime - ((((double) prgs) / currentTestMax * currentTestRuntime) + previousTestRuntime);
+
+                listener.onProgress(previousTestProgress + prgs, timeLeft);
+                break;
+            case TestAsyncTask.LOG:
+                listener.onLog(value);
+                break;
+            case TestAsyncTask.ERR:
+                listener.onError(value);
+                break;
+            case TestAsyncTask.URL:
+                listener.onUrl();
+                break;
+            case TestAsyncTask.INT:
+                listener.onInt();
+                break;
+            case TestAsyncTask.END:
+                listener.onEnd(context);
+                break;
+        }
+    }
+
+    @Override
+    public void onServiceConnected(ComponentName cname, IBinder binder) {
+        //Bind the service to this activity
+        RunTestService.TestBinder b = (RunTestService.TestBinder) binder;
+        service = b.getService();
+        isBound = true;
+        listener.onStart(service);
+        runtime = (int) Stats.of(Lists.transform(service.task.testSuites, input -> input.getRuntime(preferenceManager))).sum();
+
+    }
+
+
+    @Override
+    public void onServiceDisconnected(ComponentName name) {
+        service = null;
+    }
+
+    public boolean isBound() {
+        return isBound;
+    }
+
+    public void setBound(boolean bound) {
+        isBound = bound;
+    }
+
+    public interface EventListener {
+        void onStart(RunTestService service);
+
+        void onRun(String value);
+
+        void onProgress(int state, double eta);
+
+        void onLog(String value);
+
+        void onError(String value);
+
+        void onUrl();
+
+        void onInt();
```

Comment:
I am not super fond of the name of this method. What does it mean that the "int" event occurs? Does this mean that there is some integer? Or perhaps has this to do with interrupting? In such a case I'll say "onInterrupt".

Reply:
I was not aware of the full name so i had to make quick work of the situation